docs/Custom_Device_Tips_and_Tricks.md

```diff
@@ -0,0 +1,244 @@
+
+### Rename Protection
+
+There may be cases when you depend on a custom device item to have a certain name, and you’d like to prevent the operator from renaming the item. Add
+<DisallowRenaming>true</DisallowRenaming> below the </Name>tag for any page to prevent the operator from renaming the item.
+
+### Action VIs
+
+There are a variety of actions that can trigger a VI to run.<br />
+•	OnDelete<br />
+Executes on the deletion of a node in the system definition<br />
+•	OnLoad<br />
+Executes on the creation of a new node or load of an existing nod in the system definition<br />
+•	OnSystemShutdown<br />
+Executes on system explorer close or current system definition close<br />
+•	OnSave<br />
+Executes on save of system definition<br />
+•	OnDownload<br />
+Executes when the system definition is downloaded to the target. This VI is called after compile is complete and binary files have been created. Writing to memory should not be performed in this VI. The VI can be used to read from memory and download additional files as needed<br />
+•	OnPaste<br />
+Executes when a node is pasted within the system definition<br />
+•	OnTargetTypeChange<br />
+Executes on change of target type in the system definition<br />
+•	OnDeleteRequest<br />
+Executes on the delete request before deletion of node in system definition<br />
+•	OnCompile<br />
+Executes when the system definition is compiled during deployment. The system definition will only be compiled during deployment if there is not a good compile cache available on the host. This happens when the system definition file has been moved on disk or when changes have been made.<br />
+
+These VIs are useful if you need to make checks or perform cleanup operations after something happens. The template VIs for these actions are found in the Custom Device API library.
+
+### Run-Time Right-click Menu
+
+You can add right-click functionality in System Explorer to any custom device item. Underneath the </Item2Launch> tag for any page, add the following framework.
+
+![](images/Custom_Device_XML_Right-Click Framework.JPG)<br />
+**Custom Device XML Right-Click Framework**
+
+•	**GUID**<br />
+A unique GUID for the extra page<br />
+•	**Type_Enum**<br />
+Describes the type of right-click item<br />
+•	Action (default) runs the VI silently in the background, i.e. carry out a pre- configured task and exit<br />
+•	VI runs the VI in interactive mode displaying the front panel<br />
+•	**Execution_Enum**<br />
+•	silent runs the VI silently in the background<br />
+•	modal runs the VI as a modal window<br />
+•	floating runs the VI as a floating window<br />
+•	**Position_Enum**<br />
+•	centered (default) centers the window on the default monitor on launch<br />
+•	mouse pointer puts the font panel origin at the mouse pointer on launch<br />
+•	**Behavior_Enum**<br />
+•	None<br />
+•	OpenFrontPanel (default)<br />
+
+### Dynamic Buttons
```

Comment:
Please move this section before "### Adding shortcut menus" section as in this link: https://www.ni.com/documentation/en/veristand/latest/manual/custom-device-add-glyph-shortcut-menu-toolbar-button/

And add the data found in the "Task" column.